custom_model_runner/datarobot_drum/resource/transform_helpers.py

```diff
@@ -0,0 +1,40 @@
+import pyarrow as pa
+
+from io import BytesIO
+
+from scipy.io import mmwrite, mmread
+from scipy.sparse.csr import csr_matrix
+from scipy.sparse import vstack
+
+
+X_TRANSFORM_KEY = "X.transformed"
+
+
+def is_sparse(df):
+    return hasattr(df, "sparse") or type(df.iloc[0].values[0]) == csr_matrix
```

Comment:
why are there two conditions here?

Reply:
the first one is the case where they're actually using a pandas sparse datatype, the second is the case where they've just dumped a csr matrix into a pandas DF. we could also just enforce one pattern or another

Reply:
ok but we're deserializing the data inside of drum and this is in drum so its us using a pandas sparse df

Reply:
but it depends on what the user put in their transform hook, since that's the data structure we're serializing

Reply:
lets just land on one of them - unless there's a difference between python and R